app/services/action_queue.rb

```diff
@@ -61,8 +107,38 @@ def payload
     end
   end
 
+  class DirectDebitAction
+    include Enqueable
+    include Donatable
+
+    def payload
+      {
+        type:  'donation',
+        params: {
+          donationpage: {
+            name:             "#{@action.page.slug}-donation",
+            payment_account:  get_payment_account
+          },
+          order: {
+            amount:    data[:amount],
+            currency:  data[:currency]
+          },
+          action: {
+            source: data[:source]
+          }.merge(extra_fields_for_subscription),
+          user: user_data
+        }
+      }
+    end
+
+    def get_payment_account
+      "GoCardless #{data[:currency]}"
```

Comment:
ah I'm glad you sorted this. to check though, do we really have a GoCardless account for each currency, or is it all one account?